### PR TITLE
Feature/ui save document

### PR DESCRIPTION
**Pull request by Metaxa007**

### CODE REVIEW

File: OpenDocumentReader/DocumentViewController.swift [hunk 1]
```diff
@@ -145,6 +184,45 @@ class DocumentViewController: UIViewController, DocumentDelegate {
...
+        alert.view.layer.cornerRadius = 15
+        
+        controller.present(alert, animated: true)
+        
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + seconds) {
+            alert.dismiss(animated: true)
+            
+            completion?()
+        }
+
```
Comment (TomTasche) [inline OpenDocumentReader/DocumentViewController.swift]: blaank

**Comment by TomTasche** [comment]
Code looks good, thank you! Only very minor things where we have to agree on best practice...

Also needs some debugging of me / us to make sure everything is working as expected (see my comment on "document.save")

File: OpenDocumentReader/DocumentViewController.swift [hunk 2]
```diff
@@ -118,7 +108,46 @@ class DocumentViewController: UIViewController, DocumentDelegate {
...
+            }))
+            alert.addAction(UIAlertAction(title: "Yes", style: .default, handler: { (_) in
+                self.closeCurrentDocument() //doc.close calls autosave
+            }))
+            
+            self.present(alert, animated: true, completion: nil)
+        } else {
+            closeCurrentDocument()
+        }
+
```
[the remark this answers is not in the document]
Comment (Metaxa007) [reply OpenDocumentReader/DocumentViewController.swift]: @TomTasche fixed

**Comment by TomTasche** [comment]
Thanks for the clarifications and fixes! A few more things before we can merge and release it:

- [ ] change text of dialog: "Save it?" to "Save them now?"
- [ ] add analytics for new menu items (like https://github.com/TomTasche/OpenDocument.ios/blob/e150ac69157398cc34ad02a84007740f7feebef6/OpenDocumentReader/DocumentViewController.swift#L149)
- [ ] add analytics for new save-dialog (every time it opens + which option was chosen)
- [ ] increase time alert toast is shown (it's easy to miss in my opinion!)

General question: is the color of the alert toast if saving failed the standard color for such toasts? It really doesn't look like native iOS to me... I might be wrong though.

**Comment by Metaxa007** [comment]
@TomTasche 
Increased alert time to 1.5s. Not sure that blocking ui for too long is good for UX. 
I use a regular UIColor.red. But i agree, it looks a bit weird. From my experience the default background color of alerts is rarely changed to some other one.

**Comment by TomTasche** [comment]
As far as I can tell we are not blocking the main thread anyway because we're using asyncAfter?

Changes look good, will merge and release tomorrow!

**Comment by Metaxa007** [comment]
i mean from user perspective: during these > 1 sec he can not interact with the app, what in my opinion can have a negative effect on user experience.

**Comment by TomTasche** [comment]
Oh I see, didn't know that. 1.5s should be fine then

**Comment by TomTasche** [comment]
FYI, made one more followup commit: https://github.com/TomTasche/OpenDocument.ios/commit/84f49399eac6096641878e765ed917e25c08729a

Pushing release to App Store now.

**Pull request merged**